### DISCUSSION TITLE
DRM not removing - debug file attached

### DISCUSSION THREAD

**Discussion by Pmcc222** [noDRM/DeDRM_tools#699] [Q&A]
Hi all -

I have a series of azw3 books that I can't remove the DRM from.  I completely uninstalled Calibre, reinstalled ver 7.22.0, added DeDRM 10.0.9, and entered my eink serial numbers.  This is my 3rd kindle, so I entered all 3 serial numbers.  Tried 2 different books I have been having trouble with, and nothing.  Here is the debug file from the 2nd book:

ApplicationPaletteChange event ignored
Using calibre Qt style: True
calibre Debug log
calibre 7.22  embedded-python: True
Windows-10-10.0.19045-SP0 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.19045')
Python 3.11.5
Windows: ('10', '10.0.19045', 'SP0', 'Multiprocessor Free')
Interface language: None
EXE path: C:\Program Files\Calibre2\calibre-debug.exe
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (2, 18, 0) && DeDRM (10, 0, 9) && Package KFX (from KFX Input) (2, 18, 0) && KFX metadata reader (from KFX Input) (2, 18, 0) && From KFX (2, 18, 0) && KFX Input (2, 18, 0) && Set KFX metadata (from KFX Output) (2, 13, 0) && KFX Output (2, 13, 0) && Obok DeDRM (10, 0, 9)
calibre 7.22  embedded-python: True
Windows-10-10.0.19045-SP0 Windows ('64bit', 'WindowsPE')
('Windows', '10', '10.0.19045')
Python 3.11.5
Windows: ('10', '10.0.19045', 'SP0', 'Multiprocessor Free')
Interface language: None
EXE path: C:\Program Files\Calibre2\calibre-debug.exe
Successfully initialized third party plugins: Gather KFX-ZIP (from KFX Input) (2, 18, 0) && DeDRM (10, 0, 9) && Package KFX (from KFX Input) (2, 18, 0) && KFX metadata reader (from KFX Input) (2, 18, 0) && From KFX (2, 18, 0) && KFX Input (2, 18, 0) && Set KFX metadata (from KFX Output) (2, 13, 0) && KFX Output (2, 13, 0) && Obok DeDRM (10, 0, 9)
QPA platform: windows
devicePixelRatio: 1.5
logicalDpi: 96.0 x 96.0
physicalDpi: 138.34893617021277 x 138.54545454545453
[0.00] Starting up...
[0.02] Showing splash screen...
[1.22] splash screen shown
[1.22] Initializing db...
[1.24] db initialized
[1.24] Constructing main UI...
DEBUG:    0.0 obok::utilities.py - loading translations
DEBUG:    0.0 obok::dialogs.py - loading translations
DEBUG:    0.0 obok::config.py - loading translations
DEBUG:    0.0 obok::action_err.py - loading translations
[2.67] GUI main window shown
[2.77] main UI initialized...
[2.77] Hiding splash screen
[2.77] splash screen hidden
[2.77] Started up in 2.77 seconds with 1 books
DeDRM v10.0.9: Trying to decrypt State of Fear - Michael Crichton.azw3
stdout+stderr from file dialog helper: [b'', b'']
piped data from file dialog helper: [b'\x93\x11\'\xf5Q\xf3\xc5j\xb7\xd6/\xabk\xaac\x15"\xa8\xe6\xe5w\x10\x08k\xbe\xbb\xc9)\x100\xf5\x9c', b'C:\\Users\\User\\Documents\\New folder\\State of Fear - Michael Crichton.azw3']
MobiDeDrm v1.1.
Copyright © 2008-2022 The Dark Reverser, Apprentice Harper et al.
Decrypting Kindle Format 8 ebook: State of Fear
Found 6 keys to try after 1.0 seconds
Crypto Type is: 2
DeDRM v10.0.9: Failed to decrypt with error: No key found in 6 PIDs tried.
DeDRM v10.0.9: Looking for new default Kindle Key after 1.0 seconds
searching for kinfoFiles in C:\Users\User\AppData\Local
Found K4PC 1.25+ kinf2018 file: C:\Users\User\AppData\Local\Amazon\Kindle\storage\.kinf2018
Decrypted key file using IDString '2450186356' and UserName 'User'
DeDRM v10.0.9: Found 1 new key
MobiDeDrm v1.1.
Copyright © 2008-2022 The Dark Reverser, Apprentice Harper et al.
Decrypting Kindle Format 8 ebook: State of Fear
Got DSN key from database key_0
Found 3 keys to try after 1.3 seconds
Crypto Type is: 2
Traceback (most recent call last):
  File "calibre_plugins.dedrm.__init__", line 931, in KindleMobiDecrypt
  File "C:\Users\User\AppData\Roaming\calibre\plugins\DeDRM.zip\k4mobidedrm.py", line 203, in GetDecryptedBook
  File "C:\Users\User\AppData\Roaming\calibre\plugins\DeDRM.zip\mobidedrm.py", line 401, in processBook
mobidedrm.DrmException: No key found in 6 PIDs tried.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "calibre_plugins.dedrm.__init__", line 965, in KindleMobiDecrypt
  File "C:\Users\User\AppData\Roaming\calibre\plugins\DeDRM.zip\k4mobidedrm.py", line 203, in GetDecryptedBook
  File "C:\Users\User\AppData\Roaming\calibre\plugins\DeDRM.zip\mobidedrm.py", line 401, in processBook
mobidedrm.DrmException: No key found in 3 PIDs tried.
DeDRM v10.0.9: Ultimately failed to decrypt after 1.3 seconds. Read the FAQs at noDRM's repository: https://github.com/noDRM/DeDRM_tools/blob/master/FAQs.md
Running file type plugin DeDRM failed with traceback:
Traceback (most recent call last):
  File "calibre_plugins.dedrm.__init__", line 931, in KindleMobiDecrypt
  File "C:\Users\User\AppData\Roaming\calibre\plugins\DeDRM.zip\k4mobidedrm.py", line 203, in GetDecryptedBook
  File "C:\Users\User\AppData\Roaming\calibre\plugins\DeDRM.zip\mobidedrm.py", line 401, in processBook
mobidedrm.DrmException: No key found in 6 PIDs tried.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "calibre\customize\ui.py", line 201, in _run_filetype_plugins
  File "calibre_plugins.dedrm.__init__", line 1030, in run
  File "calibre_plugins.dedrm.__init__", line 981, in KindleMobiDecrypt
calibre_plugins.dedrm.DeDRMError: DeDRM v10.0.9: Ultimately failed to decrypt after 1.3 seconds. Read the FAQs at noDRM's repository: https://github.com/noDRM/DeDRM_tools/blob/master/FAQs.md

Any help would be appreciated!!

**Comment by ElleKayEm** [comment]
When you downloaded it, are you sure you chose one of your E-ink Kindles for the device?  Second thing to check is if that Kindle's serial number is entered correctly.  Copy and paste from the website and then restart calibre before adding the book.

**Reply by Pmcc222** [reply]
Yes, I did download it to one of my Kindles. And I copied the serial numbers directly from the Amazon website. And I did restart Calibre prior to adding the book.

**Reply by ElleKayEm** [reply]
You picked an E-ink Kindle when you downloaded it from the website to your computer, not a Fire tablet?

**Reply by Pmcc222** [reply]
Yes.  I've never had a Fire tablet.

**Reply by ElleKayEm** [reply]
Then I don't know what the issue is.  This older format should not be a problem at all.  But none of the keys you have in there are the right one for this copy of the book.
